spec/recipe.fmf

```diff
@@ -0,0 +1,87 @@
+story: |
+    As a user, I want tmt to generate a recipe file that can be reused to reproduce
+    a run with the exact same configuration, without requiring additional input.
+
+description: |
+    A recipe is a YAML file that captures a complete, static snapshot of a tmt
+    run after all dynamic evaluation has been resolved. It includes preprocessed
+    information about plans, tests, results, and run configuration. All environment
+    variables are evaluated, and git references are replaced with fixed commit hashes.
+
+    .. code-block:: yaml
+
+        # Run  configuration
+        run:
+            root: "/path/to/fmf/root"
+            remove: false
+
+        # Plan definitions with all steps
+        plans:
+            /plan/name:
+                discover:
+                    # Step data
+                    data:
+                      - name: default-0
+                        how: fmf
+                        order: 50
+                        ...
+                    # Discovered tests
+                    tests:
+                      - name: /test/name
+                        summary: Test summary
+                        ...
```

Comment:
In order to be able to execute tests the expanded test metadata will not be enough. Although we do not expect any `adjust` rules to be evaluated or any other dynamic steps, we will have to fetch the git repositories again so that the test code itself is available. Perhaps a short mention of this should be added here or in the story description?